src/Land/Model/SoilWaterParameterizations.jl

```diff
@@ -457,10 +457,7 @@ imaginary numbers, resulting in domain errors. Exit in this
 case with an error.
 """
 function effective_saturation(porosity::FT, ϑ_l::FT) where {FT}
-
-    if ϑ_l < 0
-        throw(DomainError(ϑ_l, "Effective saturation is negative."))
-    end
+    ϑ_l < 0 && error("Effective saturation is negative")
```

Comment:
@kmdeck there is a codegen issue with throwing DomainError  (and other exception types) that our version of Cassette / KernelAbstractions and CUDA can't currently inline through in later Julia versions (1.5, 1.6).  I think we'll have to address what is going on on the codegen side but for now I switched to a less specific error which also matches much of the existing core numerics code running on the GPU.

Reply:
hi! Thank you for taking care of this, and for updating the unit test. It's nice to know that a domain error on the variable \vartheta occurred (as a negative value here triggers other domain errors in certain functions, involving roots, e.g.) , but it looks like we get notified of that here still. looks good to me 👍